sigstore/_internal/ctfe.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Functionality for interacting with CT ("CTFE") signing keys.
+"""
+
+from __future__ import annotations
+
+import fnmatch
+from importlib import resources
+from typing import List
+
+import cryptography.hazmat.primitives.asymmetric.padding as padding
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec, rsa
+
+from sigstore._utils import PublicKey, key_id, load_pem_public_key
+
+
+class CTKeyringError(Exception):
+    pass
+
+
+class CTKeyring:
```

Comment:
Flagging for review: I welcome a better name for this class (and its corresponding exception class) 🙂